components/org.wso2.carbon.event.outbound.adapter.websubhub/src/main/java/org/wso2/carbon/event/outbound/adapter/websubhub/internal/util/WebSubHubEventAdapterUtil.java

```diff
@@ -0,0 +1,222 @@
+                                log.debug("Response data: " + json);
+                            } catch (IOException | ParseException e) {
+                                log.error("Error while reading WebSubHub event publisher response. " + e);
+                            }
+                        }
+                    } else {
+                        log.error("WebHubSub event publisher received " + responseCode + " code.");
+                        String jsonString;
+                        try {
+                            jsonString = EntityUtils.toString(response.getEntity());
+                            JSONParser parser = new JSONParser();
+                            JSONObject json = (JSONObject) parser.parse(jsonString);
+                            log.error("Response data: " + json);
+                        } catch (IOException | ParseException e) {
+                            log.error("Error while reading WebSubHub event publisher response. " + e);
+                        }
+                    }
+                }
+
+                @Override
+                public void failed(final Exception ex) {
+
+                    log.error("Publishing event data to WebSubHub failed. " + ex);
+                }
+
+                @Override
+                public void cancelled() {
+
+                    log.error("Publishing event data to WebSubHub cancelled.");
+                }
+            });
+        } catch (IOException | FrameworkException e) {
+            log.error("Error while publishing event data to WebSubHub. " + e);
+        }
+    }
+
+    /**
+     * Build url which is used to publish events of the given tenant domain and topic.
+     *
+     * @param topic            Topic name.
+     * @param tenantDomain     Tenant domain.
+     * @param webSubHubBaseUrl Web sub hub base url.
+     * @return Url to publish the event.
+     */
+    private static String buildURL(String topic, String tenantDomain, String webSubHubBaseUrl) {
+
+        StringBuilder tenantedTopic = new StringBuilder(tenantDomain).append(TOPIC_SEPARATOR).append(topic);
+        StringBuilder eventPublishingUrl = new StringBuilder(webSubHubBaseUrl);
+        eventPublishingUrl = eventPublishingUrl.append("?").append(HUB_MODE).append("=")
+                .append(PUBLISH).append("&").
+                append(HUB_TOPIC).append("=").append(tenantedTopic.toString());
+        return eventPublishingUrl.toString();
+    }
+
+    /**
+     * Get correlation id from the MDC.
+     * If not then generate a random UUID and return the UUID.
+     *
+     * @return Correlation id
+     */
+    public static String getCorrelationID() {
+
+        String correlationId;
+        if (isCorrelationIDPresent()) {
+            correlationId = MDC.get(CORRELATION_ID_MDC);
+        } else {
+            correlationId = UUID.randomUUID().toString();
+        }
+        return correlationId;
```

Comment:
```suggestion
        return Optional.ofNullable(MDC.get(CORRELATION_ID_MDC)).orElse(UUID.randomUUID().toString());
```

Reply:
addressed in https://github.com/wso2-extensions/identity-outbound-adapters/pull/1/commits/a26fd479692709bbd1aca48bc183978d0673f925